ra-optimisation/linear-range-action-rao/src/main/java/com/farao_community/farao/linear_range_action_rao/LinearRangeActionRao.java

```diff
@@ -35,7 +51,74 @@ public String getVersion() {
 
     @Override
     public CompletableFuture<RaoComputationResult> run(Network network, Crac crac, String variantId, ComputationManager computationManager, RaoParameters parameters) {
-        // TODO : implement LinearRangeActionRao
         return null;
     }
+
+    public CompletableFuture<RaoComputationResult> run(Network network, Crac crac, String variantId,
+                                                       ComputationManager computationManager, RaoParameters parameters,
+                                                       SensitivityComputationFactory sensitivityComputationFactory) {
+        SystematicSensitivityAnalysisResult sensiSaResults = SystematicSensitivityAnalysisService.runSensitivity(network, crac, computationManager, sensitivityComputationFactory);
+        if (sensiSaResults == null) {
+            LinearRangeActionRaoResult resultExtension = new LinearRangeActionRaoResult(LinearRangeActionRaoResult.SecurityStatus.UNSECURED);
+            RaoComputationResult raoComputationResult =  new RaoComputationResult(RaoComputationResult.Status.FAILURE);
+            raoComputationResult.addExtension(LinearRangeActionRaoResult.class, resultExtension);
+            return CompletableFuture.completedFuture(raoComputationResult);
+        }
+
+        LinearRangeActionRaoResult resultExtension = new LinearRangeActionRaoResult(LinearRangeActionRaoResult.SecurityStatus.SECURED);
+
+        // 1. do for pre
+        SensitivityComputationResults preSensi = sensiSaResults.getPrecontingencyResult();
+        List<MonitoredBranchResult> monitoredBranchResults = getMonitoredBranchResultList(crac, preSensi, resultExtension);
```

Comment:
This returns too many MonitoredBranchResult (see also comment below)
This also returns MonitoredBranchResult for CNECs which are not monitored in the N state.

--> this part should only returns 1 MonitoredBranchResult for each CNEC whose state is preventive.

Reply:
OK